pandas/core/window/rolling.py

```diff
@@ -2287,8 +2290,8 @@ def _gotitem(self, key, ndim, subset=None):
     def _validate_monotonic(self):
```

Comment:
can we just move the check on L1970 here ? (and then you can in-line L1973-1977

Reply:
I am not sure, if I understand you correctly, but we can not check for monotonic here, because non-monotonic is allowed. We have to check explicitly for nans.